pkg/worker/k8srunner.go

```diff
@@ -379,60 +403,140 @@ func (r k8sStepRunner) readLogs(ctx context.Context, podName string, opts *v1.Po
 	return scanner.Err()
 }
 
-func (r k8sStepRunner) stopPodNow(ctx context.Context, stepName, podName string) {
+func (r k8sStepRunner) stopPodNow(ctx context.Context) {
 	gracePeriod := int64(0) // 0=immediately
-	err := r.pods.Delete(ctx, podName, metav1.DeleteOptions{
+	err := r.pods.Delete(ctx, r.target.name, metav1.DeleteOptions{
 		GracePeriodSeconds: &gracePeriod,
 	})
 	if err != nil {
 		log.Warn().
 			WithError(err).
-			WithString("step", stepName).
-			WithString("pod", podName).
+			WithString("step", r.step.Name).
+			WithString("pod", r.target.name).
 			Message("Failed to delete pod.")
 	} else {
 		log.Debug().
-			WithString("step", stepName).
-			WithString("pod", podName).
+			WithString("step", r.step.Name).
+			WithString("pod", r.target.name).
 			Message("Deleted pod.")
 	}
 }
 
-func (r k8sStepRunner) continueInitContainer(podName string) error {
-	exec, err := execInPodPipeStdout(r.RestConfig, r.Config.K8s.Namespace, podName, "init", podInitContinueArgs)
+func (r k8sStepRunner) transferDataToPod(ctx context.Context) error {
+	log.Debug().WithFunc(r.logFunc).Message("Transferring repo to init container.")
+	if err := r.copyDirToPod(ctx, commonRepoVolumeMount.MountPath); err != nil {
+		return fmt.Errorf("transfer repo: %w", err)
+	}
+	log.Debug().WithFunc(r.logFunc).Message("Transferred repo to init container.")
+
+	if step, ok := r.step.Type.(wharfyml.StepDocker); ok {
+		if err := r.transferModifiedDockerfileToPod(ctx, step); err != nil {
+			return fmt.Errorf("transfer modified dockerfile: %w", err)
+		}
+
+		if err := r.copyCertToAppContainer(ctx, step); err != nil {
+			return fmt.Errorf("transfer cert: %w", err)
+		}
```

Comment:
This modified Dockerfile transfer, cert copying, and even the cert mounting should only be applied if `StepDocker.AppendCert`